x-pack/plugins/security_solution/public/management/components/artifact_entry_card/components/card_decorators/event_filters_process_descendant_indicator.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import React, { memo } from 'react';
+import { EuiSpacer, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useIsExperimentalFeatureEnabled } from '../../../../../common/hooks/use_experimental_features';
+import { isFilterProcessDescendantsEnabled } from '../../../../../../common/endpoint/service/artifacts/utils';
+import { ProcessDescendantsTooltip } from '../../../../pages/event_filters/view/components/process_descendant_tooltip';
+import type { ArtifactEntryCardDecoratorProps } from '../../artifact_entry_card';
+
+export const EventFiltersProcessDescendantIndicator = memo<ArtifactEntryCardDecoratorProps>(
+  ({ item, testIdPrefix }) => {
+    const isProcessDescendantFeatureEnabled = useIsExperimentalFeatureEnabled(
+      'filterProcessDescendantsForEventFiltersEnabled'
+    );
+
+    if (
+      isProcessDescendantFeatureEnabled &&
+      isFilterProcessDescendantsEnabled(item as ExceptionListItemSchema)
+    ) {
+      return (
+        <>
+          <EuiText
+            data-test-subj={
+              testIdPrefix !== undefined ? `${testIdPrefix}-processDescendantIndication` : undefined
```

Comment:
you can avoid doing these types of checks by using `useTestIdGenerator()` hook (this was the exact reason why it was created 😄  )

_(same comment for several other places in this PR)_

Reply:
9a768494770a7e5261d4a6c300fdce795102811a

Reply:
9a768494770a7e5261d4a6c300fdce795102811a